plugin/helpers_commands.go

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package plugin
+
+import (
+	"strings"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/pkg/errors"
+)
+
+type CommandCallback func(c *Context, args *CommandArgs) (*model.CommandResponse, *model.AppError)
+
+type CommandArgs struct {
+	Trigger      string
+	Args         []string
+	OriginalArgs *model.CommandArgs
```

Comment:
I'm not 100% happy with the name. But I can't think about a better one :wink:. If someone has ideas, it would be appreciated.

Reply:
I can't think of a better name, myself -- the only other potential idea would be to embed the original args?
```
type CommandArgs struct {
    *model.CommandArgs

    Trigger string
    Args    []string
}
```